src/pages-and-resources/discussions/DiscussionsSettings.jsx

```diff
@@ -114,7 +122,14 @@ function DiscussionsSettings({ courseId, intl }) {
               eventKey={SELECTION_STEP}
               title={intl.formatMessage(messages.providerSelection)}
             >
-              <AppList />
+              {courseDetail
```

Comment:
you are already checking `courseDetail` while calculating `canChangeProviders`, I don't think you need this here.

Reply:
`canChangeProviders` defaults to `false` so, if I don't add this check, the message will appear (and disappear later), which is potentially confusing to the user.
Changing the default to `true` is even worse because it will enable `AppList` until the course details are loaded, allowing the user to change providers.

Reply:
Actually, I've changed this so it only loads when the course details are loaded, so I can remove this.
Thanks for the suggestion.